test/integration/consul-container/libs/assert/service.go

```diff
@@ -26,6 +30,72 @@ const (
 	defaultHTTPWait    = defaultWait
 )
 
+// CatalogV2ServiceExists verifies the service name exists in the Consul catalog
+func CatalogV2ServiceExists(t *testing.T, client pbresource.ResourceServiceClient, svc string, tenancy *pbresource.Tenancy) {
+	t.Helper()
+	catalogV2ServiceHasEndpointCount(t, client, svc, tenancy, -1)
+}
+
+// CatalogV2ServiceDoesNotExist verifies the service name does not exist in the Consul catalog
+func CatalogV2ServiceDoesNotExist(t *testing.T, client pbresource.ResourceServiceClient, svc string, tenancy *pbresource.Tenancy) {
+	t.Helper()
+	ctx := testutil.TestContext(t)
+	retry.Run(t, func(r *retry.R) {
+		got, err := util.GetDecodedResource[*pbcatalog.Service](ctx, client, &pbresource.ID{
+			Type:    pbcatalog.ServiceType,
+			Name:    svc,
+			Tenancy: tenancy,
+		})
+		require.NoError(r, err, "error reading service data")
+		require.Nil(r, got, "unexpectedly found Service resource for %q", svc)
+
+		got2, err := util.GetDecodedResource[*pbcatalog.ServiceEndpoints](ctx, client, &pbresource.ID{
+			Type:    pbcatalog.ServiceEndpointsType,
+			Name:    svc,
+			Tenancy: tenancy,
+		})
+		require.NotNil(r, err, "error reading service data")
+		require.Nil(r, got2, "unexpectedly found ServiceEndpoints resource for %q", svc)
+	})
+}
+
+// CatalogV2ServiceHasEndpointCount verifies the service name exists in the Consul catalog and has the specified
+// number of workload endpoints.
+func CatalogV2ServiceHasEndpointCount(t *testing.T, client pbresource.ResourceServiceClient, svc string, tenancy *pbresource.Tenancy, count int) {
+	t.Helper()
+	require.True(t, count > 0)
+
+	catalogV2ServiceHasEndpointCount(t, client, svc, tenancy, count)
+}
+
+func catalogV2ServiceHasEndpointCount(t *testing.T, client pbresource.ResourceServiceClient, svc string, tenancy *pbresource.Tenancy, count int) {
```

Comment:
> So the xxxExists and xxxHasEndpointCount methods could share the entirety of their function bodies save for one line.

Right, but you could just do:

```suggestion
// CatalogV2ServiceHasEndpointCount verifies the service name exists in the Consul catalog and has the specified
// number of workload endpoints. A negative count checks only for the endpoint's existence.
func CatalogV2ServiceHasEndpointCount(t *testing.T, client pbresource.ResourceServiceClient, svc string, tenancy *pbresource.Tenancy, count int) {
```

Also, IMO `count == 0` is a programmer error and worth a panic, instead of `require`.